### PR TITLE
Oy2 15574 - api testing

### PR DESCRIPTION
**Pull request by daniel-belcher**
[ticket](https://qmacbis.atlassian.net/browse/OY2-15574?atlOrigin=eyJpIjoiNDExY2U0MDExYjE0NGIxMDlmY2JiZjEyZjRlNzVlOTIiLCJwIjoiaiJ9)

Adding testing and coverage reporting to app-api files and functions

#### Pull Request Creator Checklist

- [x] This PR has an associated issue or issues.
- [x] The associated issue(s) are linked above.
- [x] This PR meets all acceptance criteria for those issues.
- [x] This PR and linked issue(s) are adequately documented
- [x] This PR and linked issues(s) are a complete description of the changeset; an individual or team should be able to understand the issue(s) and changes by reading through this PR and it's links, with no further interaction.
- [x] Someone has been assigned this PR.
- [x] At least one person has been marked as reviewer on this PR.

#### Pull Request Reviewer/Assignee Checklist

- [ ] This PR has an associated issue or issues.
- [ ] The associated issue(s) are linked above.
- [ ] This PR meets all acceptance criteria for those issues.
- [ ] This PR and linked issue(s) are adequately documented
- [ ] This PR and linked issues(s) are a complete description of the changeset; an individual or team should be able to understand the issue(s) and changes by reading through this PR and it's links, with no further interaction.

### CODE REVIEW

File: services/app-api/libs/dynamodb-lib.ts [hunk 1]
```diff
@@ -1,30 +1,48 @@
...
+    return { ...result, Item: result?.Item as Result | undefined };
+  },
+  put: (params: DynamoCreate) => client.put(params).promise(),
+  post: (params: DynamoCreate) => client.put(params).promise(),
+  scan: async <Result = CoreSet | Measure>(params: DynamoScan) => {
+    const result = await client.scan(params).promise();
+    return { ...result, Items: result?.Items as Result[] | undefined };
+  },
+  update: (params: DynamoUpdate) => client.update(params).promise(),
+  delete: (params: DynamoDelete) => client.delete(params).promise(),
```
Comment (benjaminpaige) [inline services/app-api/libs/dynamodb-lib.ts:44]: 👍  awesome

File: services/ui-src/src/views/StateHome/index.tsx [hunk 2]
```diff
@@ -73,6 +73,7 @@ export const StateHome = () => {
   const { state, year } = useParams();
   const queryClient = useQueryClient();
   const { data, error, isLoading } = Api.useGetCoreSets();
+  console.log("data", data);
```
Comment (benjaminpaige) [inline services/ui-src/src/views/StateHome/index.tsx:76]: do we need these logs?

File: services/ui-src/src/views/CoreSet/index.tsx [hunk 3]
```diff
@@ -91,6 +91,7 @@ const QualifiersStatusAndLink = ({ coreSetId }: { coreSetId: CoreSetAbbr }) => {
 const useMeasureTableDataBuilder = () => {
   const { state, year, coreSetId } = useParams();
   const { data, isLoading, isError, error } = useGetMeasures();
+  console.log("measures", data);
```
Comment (benjaminpaige) [inline services/ui-src/src/views/CoreSet/index.tsx:94]: .

File: services/ui-src/src/views/StateHome/index.tsx (hunk 2 above)
Comment (daniel-belcher) [reply services/ui-src/src/views/StateHome/index.tsx:76]: Honestly forgot they were added
they were their for some early debug

**Review by benjaminpaige — APPROVED** [review]
Looks great! love that we are throwing and catching some errors now, even if they are returning 500's - thanks Dan!

**Review by benjaminpaige — APPROVED** [review]

**Pull request merged**